### PR TITLE
Home Alias fixes for issue #5791

### PR DESCRIPTION
**Pull request by sfmskywalker**
- Fixes a bug with the HomeAlias service
- Fixes Pages specflow test (by changing the specification)

### CODE REVIEW

File: src/Orchard.Web/Modules/Orchard.Alias/Implementation/Map/AliasMap.cs [hunk 1]
```diff
@@ -171,7 +171,8 @@ public class AliasMap {
             }
 
             if (match == null) {
-                var foundPath = focus.Paths.Keys.FirstOrDefault();
+                // Get the shortest path available to ensure the "home" alias is always taken if present.
+                var foundPath = focus.Paths.Keys.OrderBy(x => x).FirstOrDefault();
```
Comment (Jetski5822) [inline src/Orchard.Web/Modules/Orchard.Alias/Implementation/Map/AliasMap.cs]: I feel like OrderBy Key isent the right thing. what if my alias a length of 1 and wasent / ?

File: src/Orchard.Web/Modules/Orchard.Autoroute/Services/HomeAliasService.cs [hunk 2]
```diff
@@ -19,21 +24,27 @@ public class HomeAliasService : IHomeAliasService {
...
 
-        public int? GetHomePageId() {
-            var homePageRoute = GetHomeRoute();
-            var homePageIdValue = homePageRoute != null && homePageRoute.ContainsKey("id") ? (string)homePageRoute["id"] : default(string);
-            var homePageId = TryParseInt32(homePageIdValue);
-
-            return homePageId;
+        public int? GetHomePageId(VersionOptions version = null) {
+            var homePage = GetHomePage(version);
+            return homePage != null ? homePage.Id : default(int?);
```
Comment (Jetski5822) [inline src/Orchard.Web/Modules/Orchard.Autoroute/Services/HomeAliasService.cs]: much cleaner

File: src/Orchard.Web/Modules/Orchard.Alias/Implementation/Map/AliasMap.cs (hunk 1 above)
Comment (sfmskywalker) [inline src/Orchard.Web/Modules/Orchard.Alias/Implementation/Map/AliasMap.cs]: Good catch. It needs to order by the length of the key, not the value of the key.

File: src/Orchard.Web/Modules/Orchard.Autoroute/Services/HomeAliasService.cs (hunk 2 above)
Comment (sfmskywalker) [inline src/Orchard.Web/Modules/Orchard.Autoroute/Services/HomeAliasService.cs]: Thanks. And equally important, it fixes the issue for content items that use a custom route that doesn't involve the "id" route value (such as Blogs which uses "blogId").

File: src/Orchard.Web/Modules/Orchard.Autoroute/Services/IHomeAliasService.cs [hunk 3]
```diff
@@ -5,8 +5,9 @@ namespace Orchard.Autoroute.Services {
     
     public interface IHomeAliasService : IDependency {
         RouteValueDictionary GetHomeRoute();
-        int? GetHomePageId();
+        int? GetHomePageId(VersionOptions version = null);
```
Comment (sebastienros) [inline src/Orchard.Web/Modules/Orchard.Autoroute/Services/IHomeAliasService.cs]: In case we keep the class, we can remove it.

**Pull request merged**